CMakeLists.txt

```diff
@@ -511,18 +511,27 @@ if(NOT AVIF_CODEC_AOM_ENABLED
     message(WARNING "libavif: No decoding library is enabled.")
 endif()
 
-set_target_properties(avif PROPERTIES VERSION ${LIBRARY_VERSION} SOVERSION ${LIBRARY_SOVERSION} C_VISIBILITY_PRESET hidden)
-target_include_directories(avif PUBLIC $<BUILD_INTERFACE:${libavif_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include>)
+set_target_properties(avif_obj PROPERTIES C_VISIBILITY_PRESET hidden)
+set_property(TARGET avif_obj PROPERTY POSITION_INDEPENDENT_CODE ON)
```

Comment:
This doesn't seem to be in the original code. This is a good change. I am just wondering if it needs to be part of this pull request.

If this change can stand alone, it is better to do it separately.

Reply:
This is due to the fact that avif_obj is now kindof like a static library. So, even if you build a shared lib, you need that. I set CMAKE_POSITION_INDEPENDENT_CODE once and for all at the beginning.